packages/@aws-cdk/aws-lambda-golang/README.md

```diff
@@ -0,0 +1,246 @@
+   If you are not vendoring then `go build` will be run without `-mod=vendor`
+    since the default behavior is to download dependencies
+
+All other properties of `lambda.Function` are supported, see also the [AWS Lambda construct library](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda).
+
+## Environment
+
+By default the following environment variables are set for you:
+
+* `GOOS=linux`
+* `GOARCH=amd64`
+* `GO111MODULE=on`
+
+Use the `environment` prop to define additional environment variables when go runs:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    environment: {
+      HELLO: 'WORLD',
+    },
+  },
+});
+```
+
+## Docker
+
+To force bundling in a docker container, set the `forceDockerBundling` prop to `true`. This
+
+Use the `buildArgs` prop to pass build arguments when building the bundling image:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    buildArgs: {
+      HTTPS_PROXY: 'https://127.0.0.1:3001',
+    },
+  },
+});
+```
+
+Use the `bundling.dockerImage` prop to use a custom bundling image:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    dockerImage: cdk.DockerImage.fromBuild('/path/to/Dockerfile'),
+  },
+});
+```
+
+Use the `bundling.goBuildFlags` prop to pass additional build flags to `go build`:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    goBuildFlags: ['-ldflags "-s -w"'],
+  },
+});
+```
+
+## Command hooks
+
+It is  possible to run additional commands by specifying the `commandHooks` prop:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  bundling: {
+    commandHooks: {
+      // run tests
+      beforeBundling(inputDir: string): string[] {
+        return ['go test ./cmd/api -v'];
+      },
+      // ...
+    },
+  },
+});
+```
+
+The following hooks are available:
+
+* `beforeBundling`: runs before all bundling commands
+* `afterBundling`: runs after all bundling commands
+
+They all receive the directory containing the `go.mod` file (`inputDir`) and the
+directory where the bundled asset will be output (`outputDir`). They must return
+an array of commands to run. Commands are chained with `&&`.
+
+The commands will run in the environment in which bundling occurs: inside the
+container for Docker bundling or on the host OS for local bundling.
+
+## Additional considerations
+
+Depending on how you structure your Golang application, you may want to change the `assetHashType` parameter.
+By default this parameter is set to `AssetHashType.OUTPUT` which means that the CDK will calculate the asset hash
+(and determine whether or not your code has changed) based on the Golang executable that is created.
+
+If you specify `AssetHashType.SOURCE`, the CDK will calculate the asset hash by looking at the folder
```

Comment:
Any reason not to use SOURCE as  the default?

It seems to me that using SOURCE as the default is a better choice for go, while requiring that `go.sum` is included. The benefit of using SOURCE is that it can skip the _build_ if the source did not change.